src/window.rs

```diff
@@ -126,8 +126,11 @@ impl WindowWrapper {
 
     pub fn handle_key_down(&mut self, keycode: Keycode, modifiers: Mod) {
         trace!("KeyDown Received: {}", keycode);
+        const ALTGR: Mod = Mod::from_bits_truncate(0x0240);
```

Comment:
Could you try https://docs.rs/sdl2/0.33.0/sdl2/keyboard/struct.Mod.html#associatedconstant.MODEMOD instead of using the from bits truncate path? I think it represents the AltGR key like you are doing

Reply:
https://github.com/Rust-SDL2/rust-sdl2/blob/73afe22a08443d6e494e24ed4b8133ca868b2f2e/src/sdl2/keyboard/mod.rs#L28 welp maybe not. Docs say it should, but it looks different...

Reply:
I thought it'd be that too, but MODEMOD seem to correspond to 0x4000 and not 0x0240

Reply:
0x0240 would be RALTMOD and LCTRLMOD together

Reply:
Does AltGr only trigger if it is the Left Ctrl and the Right Alt? If you hold Left Alt and Right Ctrl does it do the same thing?

Reply:
Yes, I can accomplish the same thing with Left Alt and Right Ctrl as with Alt Gr when writing in VS Code for example.

Reply:
It seems any combination of ALT and CTRL will work as Alt Gr.

Reply:
ok. Heres my proposal then:
```
let ctrl = modifiers.contains(Mod::LCTRLMOD) || modifiers.contains(Mod::RCTRLMOD);
let alt = modifiers.contains(Mod::LALTMOD) || modifiers.contains(Mod::RALTMOD);
let gui = modifiers.contains(Mod::LGUIMOD) || modifiers.contains(Mod::RGUIMOD);
let will_text_input = 
    modifiers.contains(Mod::MODEMOD) ||
    (ctrl && alt) ||
    !ctrl && !alt && !gui;
```
Thoughts?

Reply:
I will try it out! 😄

Reply:
For extra explicit-ness we could define altGr to equal ctrl && alt, but I could take that change or leave it

Reply:
It seems to be working for all the combinations that I could think of. Should I update this draft with that?

Reply:
Yes! And I'll get it merged